openupgrade_scripts/scripts/sale_stock/17.0.1.0/post-migration.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2025 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    # Fill delivery_date and incoterm_location according
+    # https://github.com/odoo/odoo/blob/00818b7bfaf22635b0c40b3b9c7e37e0c9789da1/
+    # addons/sale_stock/models/account_move.py#L116-L134
+    openupgrade.logged_query(
+        env.cr,
+        """
+        WITH sub AS (
+            SELECT
+                MAX(so.effective_date) as effective_date,
+                MAX(COALESCE(so.incoterm_location, '')) as incoterm_location,
+                am.id as move_id
+            FROM sale_order so
+            JOIN sale_order_line sol ON sol.order_id = so.id
+            JOIN sale_order_line_invoice_rel rel ON rel.order_line_id = sol.id
+            JOIN account_move_line aml ON rel.invoice_line_id = aml.id
+            JOIN account_move am ON aml.move_id = am.id
+            GROUP BY am.id
```

Comment:
```suggestion
            WHERE am.state != 'cancel'
            GROUP BY am.id
```

Reply:
I don't think so. The date and incoterm are filled even if the invoices are canceled. They may come back to draft in any moment, and you want that information there.